yarn-project/noir-compiler/src/compile/noir/noir-wasm-compiler.ts

```diff
@@ -0,0 +1,133 @@
+  /** Logging function */
+  log: LogFn;
+  /** Log debugging information through this function */
+  debugLog?: LogFn;
+};
+
+/**
+ * Noir Package Compiler
+ */
+export class NoirWasmContractCompiler {
+  #log: LogFn;
+  #debugLog: LogFn;
+  #package: NoirPackage;
+  #fm: FileManager;
+  #dependencyManager: NoirDependencyManager;
+
+  private constructor(
+    entrypoint: NoirPackage,
+    dependencyManager: NoirDependencyManager,
+    fileManager: FileManager,
+    opts: NoirWasmCompileOptions,
+  ) {
+    this.#log = opts.log;
+    this.#debugLog = opts.debugLog ?? createDebugLogger('noir-compiler:wasm');
+    this.#package = entrypoint;
+    this.#fm = fileManager;
+    this.#dependencyManager = dependencyManager;
+  }
+
+  /**
+   * Creates a new compiler instance.
+   * @param fileManager - The file manager to use
+   * @param projectPath - The path to the project
+   * @param opts - Compilation options
+   */
+  public static new(fileManager: FileManager, projectPath: string, opts: NoirWasmCompileOptions) {
+    if (!isAbsolute(projectPath)) {
+      throw new Error('projectPath must be an absolute path');
+    }
+
+    const noirPackage = NoirPackage.open(projectPath, fileManager);
+    if (noirPackage.getType() !== 'contract') {
+      throw new Error('This is not a contract project');
+    }
+
+    const dependencyManager = new NoirDependencyManager(
+      [
+        new LocalDependencyResolver(fileManager),
+        new GithubCodeArchiveDependencyResolver(fileManager),
+        // TODO support actual Git repositories
+      ],
+      noirPackage,
+    );
+
+    return new NoirWasmContractCompiler(noirPackage, dependencyManager, fileManager, opts);
+  }
+
+  /**
+   * Compiles the project.
+   */
+  public async compile(): Promise<NoirCompilationArtifacts[]> {
+    this.#debugLog(`Compiling contract at ${this.#package.getEntryPointPath()}`);
+    await this.#dependencyManager.resolveDependencies();
+    this.#debugLog(`Dependencies: ${this.#dependencyManager.getPackageNames().join(', ')}`);
+
+    initializeResolver(this.#resolveFile);
+
+    try {
+      const contract = await compile(this.#package.getEntryPointPath(), true, {
+        /* eslint-disable camelcase */
+        root_dependencies: this.#dependencyManager.getEntrypointDependencies(),
+        library_dependencies: this.#dependencyManager.getLibraryDependencies(),
+        /* eslint-enable camelcase */
+      });
+
+      return [{ contract }];
+    } catch (err) {
+      if (err instanceof Error && err.name === 'CompileError') {
+        this.#processCompileError(err as CompileError);
+      } else {
+        this.#log('Error compiling contract ' + err);
+      }
+
+      throw new Error("Couldn't compile contract");
+    }
+  }
+
+  #resolveFile = (path: string) => {
+    try {
+      const libFile = this.#dependencyManager.findFile(path);
+      return this.#fm.readFileSync(libFile ?? path, 'utf-8');
+    } catch (err) {
+      return '';
```

Comment:
Shouldn't we log or throw an error here? Or is it handled by the caller?

Reply:
I was looking at the default implementation and it seemed to return an empty string and I assumed it's just because wasm couldn't handle it?

https://github.com/noir-lang/noir/blob/dbb6cf008dadc406e62ea82c5ec98fdc461eb76a/compiler/source-resolver/src/index_node.ts#L5-L14

Reply:
Fair point! Let's at least log it though.